services/types.go

```diff
@@ -86,6 +87,34 @@ type Services interface {
 	LeaderService(service ServiceID, opts ElectionOptions) (LeaderService, error)
 }
 
+// Watch is a watcher that issues notification when a service is updated
+type Watch interface {
+	xclose.SimpleCloser
+
+	// C returns the notification channel
+	C() <-chan struct{}
+
+	// Get returns the latest service of the service watchable
+	Get() Service
+}
+
+// Watchable is a container of a service.Service value
+// that can be subscribed to by issuing a watch
+type Watchable interface {
```

Comment:
Can skip the Service specific watchable and only expose ServiceWatch? I think internally using a xwatch.Watch should be safe enough?